### PR TITLE
enhance:[cherry-pick] Support calculate segment maxRows with diskann

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #34495 

master pr: #35076

### CODE REVIEW

File: internal/datacoord/compaction_trigger_v2.go [hunk 1]
```diff
@@ -393,6 +399,37 @@ func (m *CompactionTriggerManager) SubmitSingleViewToScheduler(ctx context.Conte
...
 
+func (m *CompactionTriggerManager) getExpectedSegmentSize(collectionID int64) int64 {
+	indexInfos := m.meta.indexMeta.GetIndexesForCollection(collectionID, "")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	collMeta, err := m.handler.GetCollection(ctx, collectionID)
+	if err != nil {
+		log.Warn("failed to get collection", zap.Int64("collectionID", collectionID), zap.Error(err))
+		return Params.DataCoordCfg.SegmentMaxSize.GetAsInt64() * 1024 * 1024
```
Comment (wayblink) [inline internal/datacoord/compaction_trigger_v2.go:410]: It supposed not happen in normal？will be blocked even though return a value here？

Comment (xiaocai2333) [reply internal/datacoord/compaction_trigger_v2.go:410]: yeah， you are right, it should return an error.

**Pull request merged**